Heroic runs into ENOENT on fedora images

Here are the logs of the heroic container with image tag :fedora

```
2026-04-29 01:14:11] [ /etc/cont-init.d/10-setup_user.sh: executing... ]
[2026-04-29 01:14:11] **** Configure default user ****
[2026-04-29 01:14:11] Setting default user uid=1000(retro) gid=1000(retro)
useradd: warning: the home directory /home/retro already exists.
useradd: Not copying any file from skel directory into it.
[2026-04-29 01:14:11] Setting umask to 000
[2026-04-29 01:14:11] Ensure retro home directory is writable
[2026-04-29 01:14:11] Ensure XDG_RUNTIME_DIR is writable
[2026-04-29 01:14:11] DONE
[2026-04-29 01:14:11] 
[2026-04-29 01:14:11] [ /etc/cont-init.d/15-setup_devices.sh: executing... ]
[2026-04-29 01:14:11] **** Configure devices ****
[2026-04-29 01:14:11] Exec device groups
[2026-04-29 01:14:11] Path '/dev/input/*' is not present.
[2026-04-29 01:14:11] Path '/dev/nvidia*' is not present.
[2026-04-29 01:14:11] Adding user 'retro' to groups: video,render
[2026-04-29 01:14:11] DONE
[2026-04-29 01:14:11] 
[2026-04-29 01:14:11] [ /etc/cont-init.d/30-nvidia.sh: executing... ]
[2026-04-29 01:14:11] 
[2026-04-29 01:14:11] [ /etc/cont-init.d/init-gamescope.sh: executing... ]
[2026-04-29 01:14:11] Launching the container's startup script as user 'retro'
[2026-04-29 01:14:11] [Sway] - Starting: `/usr/bin/heroic --no-sandbox`
2026-04-29 01:14:11 - [main.c:328] Could not find config for output WL-1 ((null))
Failed to connect to user scope bus via local transport: No such file or directory
Failed to connect to user scope bus via local transport: No such file or directory
Failed to connect to user scope bus via local transport: No such file or directory
Failed to connect to user scope bus via local transport: No such file or directory
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activating service name='org.a11y.Bus' requested by ':1.2' (uid=1000 pid=87 comm="waybar -b bar-0" label="system_u:system_r:spc_t:s0")
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Successfully activated service 'org.a11y.Bus'
[92:0429/011411.272990:ERROR:dbus/bus.cc:405] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2026-04-29 01:14:11.276] [info] Using configuration file /home/retro/.config/waybar/config.jsonc
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activating service name='org.freedesktop.portal.Desktop' requested by ':1.1' (uid=1000 pid=87 comm="waybar -b bar-0" label="system_u:system_r:spc_t:s0")
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activating service name='org.freedesktop.impl.portal.PermissionStore' requested by ':1.4' (uid=1000 pid=123 comm="/usr/libexec/xdg-desktop-portal" label="system_u:system_r:spc_t:s0")
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Successfully activated service 'org.freedesktop.impl.portal.PermissionStore'
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activating service name='org.freedesktop.portal.Documents' requested by ':1.4' (uid=1000 pid=123 comm="/usr/libexec/xdg-desktop-portal" label="system_u:system_r:spc_t:s0")
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Successfully activated service 'org.freedesktop.portal.Documents'
fuse: device not found, try 'modprobe fuse' first
error: fuse init failed: Can't mount path /run/user/wolf/doc

(/usr/libexec/xdg-desktop-portal:123): xdg-desktop-portal-WARNING **: 01:14:11.289: Document portal fuse mount point unknown: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name :1.6 was not provided by any .service files

(/usr/libexec/xdg-desktop-portal:123): xdg-desktop-portal-WARNING **: 01:14:11.297: Failed to create RealtimeKit proxy: Could not connect: No such file or directory
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activating service name='org.freedesktop.impl.portal.desktop.gtk' requested by ':1.4' (uid=1000 pid=123 comm="/usr/libexec/xdg-desktop-portal" label="system_u:system_r:spc_t:s0")
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activating service name='org.freedesktop.systemd1' requested by ':1.7' (uid=1000 pid=88 comm="/usr/bin/python3 /usr/libexec/sway-systemd/assign-" label="system_u:system_r:spc_t:s0")
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
ERROR:assign-cgroups:DBus connection error: Process org.freedesktop.systemd1 exited with status 1
dbus-daemon[115]: Activating service name='org.a11y.atspi.Registry' requested by ':1.1' (uid=1000 pid=144 comm="/usr/libexec/xdg-desktop-portal-gtk" label="system_u:system_r:spc_t:s0")

(xdg-desktop-portal-gtk:144): xdg-desktop-portal-gtk-WARNING **: 01:14:11.331: error: Could not connect: No such file or directory

dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Successfully activated service 'org.freedesktop.impl.portal.desktop.gtk'
dbus-daemon[115]: Successfully activated service 'org.a11y.atspi.Registry'
SpiRegistry daemon is running with well-known name - org.a11y.atspi.Registry

(/usr/libexec/xdg-desktop-portal:123): xdg-desktop-portal-WARNING **: 01:14:11.339: Failed connect to PipeWire: Couldn't connect to PipeWire
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Successfully activated service 'org.freedesktop.portal.Desktop'
[2026-04-29 01:14:11.343] [info] Using CSS file /home/retro/.config/waybar/style.css
[2026-04-29 01:14:11.347] [warning] Unable to connect to the SYSTEM Bus!...
[2026-04-29 01:14:11.348] [warning] 'swap-icon-label' must be a bool.
[2026-04-29 01:14:11.361] [error] Can't open RFKILL control device
[2026-04-29 01:14:11.481] [warning] Requested height: 30 is less than the minimum height: 35 required by the modules
[2026-04-29 01:14:11.481] [info] Bar configured (width: 3456, height: 35) for output: WL-1
[92:0429/011411.872125:ERROR:dbus/bus.cc:405] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[92:0429/011411.872156:ERROR:dbus/bus.cc:405] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
(01:14:11) [INFO]:    Applying migration "legendary-move-global-config-folder"
(01:14:11) [ERROR]:   [Backend]:         Error: spawn free ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections (node:internal/process/task_queues:90:21)
(01:14:11) [INFO]:    Migration "legendary-move-global-config-folder" successfully applied
(01:14:11) [INFO]:    [Connection]:      Connectivity: check-online
(01:14:11) [INFO]:    [Connection]:      Pinging external endpoints
(01:14:11) [INFO]:    [Gog]:             Checking for existing gog manifests
(node:92) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `heroic --trace-deprecation ...` to show where the warning was created)
Checking for update
The XKEYBOARD keymap compiler (xkbcomp) reports:
> Warning:          Could not resolve keysym XF86OK
> Warning:          Could not resolve keysym XF86GoTo
> Warning:          Could not resolve keysym XF86VendorLogo
> Warning:          Could not resolve keysym XF86MediaSelectProgramGuide
> Warning:          Could not resolve keysym XF86MediaSelectProgramGuide
> Warning:          Could not resolve keysym XF86MediaSelectHome
> Warning:          Could not resolve keysym XF86MediaLanguageMenu
> Warning:          Could not resolve keysym XF86MediaTitleMenu
> Warning:          Could not resolve keysym XF86AudioChannelMode
> Warning:          Could not resolve keysym XF86MediaSelectPC
> Warning:          Could not resolve keysym XF86MediaSelectTV
> Warning:          Could not resolve keysym XF86MediaSelectCable
> Warning:          Could not resolve keysym XF86MediaSelectVCR
> Warning:          Could not resolve keysym XF86MediaSelectVCRPlus
> Warning:          Could not resolve keysym XF86MediaSelectSatellite
> Warning:          Could not resolve keysym XF86MediaSelectCD
> Warning:          Could not resolve keysym XF86MediaSelectTape
> Warning:          Could not resolve keysym XF86MediaSelectRadio
> Warning:          Could not resolve keysym XF86MediaSelectTuner
> Warning:          Could not resolve keysym XF86MediaPlayer
> Warning:          Could not resolve keysym XF86MediaSelectTeletext
> Warning:          Could not resolve keysym XF86MediaSelectAuxiliary
> Warning:          Could not resolve keysym XF86MediaPlaySlow
> Warning:          Could not resolve keysym XF86NumberEntryMode
> Warning:          Could not resolve keysym XF86RefreshRateToggle
> Warning:          Could not resolve keysym XF86Accessibility
> Warning:          Could not resolve keysym XF86DoNotDisturb
Errors from xkbcomp are not fatal to the X server
The XKEYBOARD keymap compiler (xkbcomp) reports:
> Warning:          Unsupported maximum keycode 708, clipping.
>                   X11 cannot support keycodes above 255.
> Warning:          Virtual modifier Hyper multiply defined
>                   Using 0, ignoring 0
> Warning:          Virtual modifier ScrollLock multiply defined
>                   Using 0, ignoring 0
> Warning:          Could not resolve keysym XF86OK
> Warning:          Could not resolve keysym XF86GoTo
> Warning:          Could not resolve keysym XF86VendorLogo
> Warning:          Could not resolve keysym XF86MediaSelectProgramGuide
> Warning:          Could not resolve keysym XF86MediaSelectProgramGuide
> Warning:          Could not resolve keysym XF86MediaSelectHome
> Warning:          Could not resolve keysym XF86MediaLanguageMenu
> Warning:          Could not resolve keysym XF86MediaTitleMenu
> Warning:          Could not resolve keysym XF86AudioChannelMode
> Warning:          Could not resolve keysym XF86MediaSelectPC
> Warning:          Could not resolve keysym XF86MediaSelectTV
> Warning:          Could not resolve keysym XF86MediaSelectCable
> Warning:          Could not resolve keysym XF86MediaSelectVCR
> Warning:          Could not resolve keysym XF86MediaSelectVCRPlus
> Warning:          Could not resolve keysym XF86MediaSelectSatellite
> Warning:          Could not resolve keysym XF86MediaSelectTape
> Warning:          Could not resolve keysym XF86MediaSelectRadio
> Warning:          Could not resolve keysym XF86MediaSelectTuner
> Warning:          Could not resolve keysym XF86MediaPlayer
> Warning:          Could not resolve keysym XF86MediaSelectTeletext
> Warning:          Could not resolve keysym XF86MediaSelectAuxiliary
> Warning:          Could not resolve keysym XF86MediaPlaySlow
> Warning:          Could not resolve keysym XF86NumberEntryMode
> Warning:          Could not resolve keysym XF86RefreshRateToggle
> Warning:          Could not resolve keysym XF86Accessibility
> Warning:          Could not resolve keysym XF86DoNotDisturb
Errors from xkbcomp are not fatal to the X server
(01:14:12) [WARNING]: [Backend]:         Protocol already registered.
[92:0429/011412.117673:ERROR:content/browser/browser_main_loop.cc:290] Gtk: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
Generated new staging user ID: a19a920a-2798-5cda-960a-243862f70315
(01:14:12) [INFO]:    [Connection]:      Connectivity: online
(01:14:12) [INFO]:    [Backend]:         User Not Found, removing it from Store
(01:14:12) [ERROR]:   [Gog]:             Unable to syncQueued playtime, userData not present
(01:14:12) [DEBUG]:   [Backend]:         Logging to file(s) /home/retro/.local/state/Heroic/logs/runners/gog.log
(01:14:12) [INFO]:    [Gog]:             Running command: GOGDL_CONFIG_PATH=/home/retro/.config/heroic/gogdlConfig /opt/Heroic/resources/app.asar.unpacked/build/bin/x64/linux/gogdl --auth-config-path /home/retro/.config/heroic/gog_store/auth.json auth
(01:14:12) [DEBUG]:   [Backend]:         Releases info ready, checking wine releases
(01:14:12) [DEBUG]:   [Backend]:         Releases info ready, checking anticheat data
(01:14:12) [INFO]:    [Backend]:         Frontend Ready
Error: Error: net::ERR_NETWORK_CHANGED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:134240)
    at SimpleURLLoaderWrapper.emit (node:events:508:28)
(01:14:12) [ERROR]:   [Backend]:         failed to update Error: net::ERR_NETWORK_CHANGED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:134240)
    at SimpleURLLoaderWrapper.emit (node:events:508:28)
(node:92) UnhandledPromiseRejectionWarning: Error: net::ERR_NETWORK_CHANGED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:134240)
    at SimpleURLLoaderWrapper.emit (node:events:508:28)
(node:92) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 6)
(01:14:12) [INFO]:    [Backend]:         AreWeAntiCheatYet data downloaded
(01:14:12) [INFO]:    [Backend]:         Checking for new Heroic Updates
(01:14:12) [INFO]:    [Backend]:         Checking for current version changelog
(01:14:12) [INFO]:    [Frontend]:        Updated to a new version, cleaaning up the cache.
(01:14:12) [DEBUG]:   [Backend]:         Logging to file(s) /home/retro/.local/state/Heroic/logs/runners/legendary.log
(01:14:12) [INFO]:    [Legendary]:       Running command: LEGENDARY_CONFIG_PATH=/home/retro/.config/heroic/legendaryConfig/legendary /opt/Heroic/resources/app.asar.unpacked/build/bin/x64/linux/legendary cleanup
(01:14:12) [INFO]:    [Frontend]:        Refreshing undefined Library
(01:14:12) [WARNING]: refresh not implemented on Sideload Library Manager
(01:14:12) [INFO]:    [Legendary]:       Refreshing library...
(01:14:14) [INFO]:    [WineDownloader]:  Updating wine versions info
(01:14:14) [INFO]:    [WineDownloader]:  Fetching upstream information...
(01:14:15) [INFO]:    [WineDownloader]:  wine versions updated
(01:14:15) [DEBUG]:   [Backend]:         Unable to load Steam Libraries, libraryfolders.vdf not found
(01:14:15) [INFO]:    [Winetricks]:      Downloading Winetricks
(01:14:15) [INFO]:    [ToolInstaller]:   Updating dxvk to: dxvk-2.7.1
(01:14:15) [INFO]:    [ToolInstaller]:   Updating vkd3d to: vkd3d-proton-2.14.1
(01:14:15) [INFO]:    [ToolInstaller]:   Updating dxvk-nvapi to: dxvk-nvapi-v0.9.1
(01:14:16) [INFO]:    [DownloadManager]: Downloader: Finished downloading https://github.com/Heroic-Games-Launcher/vkd3d-proton/releases/download/v2.14.1/vkd3d-proton-2.14.1.tar.xz
(01:14:16) [INFO]:    [ToolInstaller]:   Downloaded vkd3d, extracting...
(01:14:16) [INFO]:    [DownloadManager]: Downloader: Finished downloading https://github.com/jp7677/dxvk-nvapi/releases/download/v0.9.1/dxvk-nvapi-v0.9.1.tar.gz
(01:14:16) [INFO]:    [ToolInstaller]:   Downloaded dxvk-nvapi, extracting...
(01:14:16) [INFO]:    [ToolInstaller]:   vkd3d updated!
(01:14:16) [INFO]:    [ToolInstaller]:   dxvk-nvapi updated!
(01:14:16) [INFO]:    [DownloadManager]: Downloader: Finished downloading https://github.com/doitsujin/dxvk/releases/download/v2.7.1/dxvk-2.7.1.tar.gz
(01:14:16) [INFO]:    [ToolInstaller]:   Downloaded dxvk, extracting...
(01:14:16) [INFO]:    [ToolInstaller]:   dxvk updated!
(01:14:17) [INFO]:    [Backend]:         Starting the Download Queue
(node:92) [DEP0187] DeprecationWarning: Passing invalid argument types to fs.existsSync is deprecated
00:00:38.283 [ERROR] [wlr] [render/swapchain.c:93] No free output buffer slot
00:01:12.717 [ERROR] [wlr] [render/swapchain.c:93] No free output buffer slot
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activating service name='org.freedesktop.systemd1' requested by ':1.18' (uid=1000 pid=411 comm="kitty" label="system_u:system_r:spc_t:s0")
dbus-daemon[56]: [session uid=1000 pid=56 pidfd=5] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
[0.158] Could not move child process into a systemd scope: [Errno 5] Failed to call StartTransientUnit: org.freedesktop.DBus.Error.Spawn.ChildExited: Process org.freedesktop.systemd1 exited with status 1
[0.197] [glfw error 65544]: Notify: Failed to get server capabilities error: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files
(01:17:37) [ERROR]:   [Backend]:         Error: spawn free ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections (node:internal/process/task_queues:90:21)
```

--- a/images/base-app/build-fedora/Dockerfile
+++ b/images/base-app/build-fedora/Dockerfile
@@ -44,7 +44,7 @@ ARG REQUIRED_PACKAGES="\
     gamescope \
     sway \
     xorg-x11-server-Xwayland kitty nano \
-    waybar default-fonts xdg-desktop-portal xdg-desktop-portal-gtk psmisc \
+    waybar mako default-fonts xdg-desktop-portal xdg-desktop-portal-gtk psmisc \
     mangohud \
     sdl2-compat ncurses \
     dbus-daemon \

--- a/images/base-app/build-fedora/Dockerfile
+++ b/images/base-app/build-fedora/Dockerfile
@@ -47,6 +47,7 @@ ARG REQUIRED_PACKAGES="\
     waybar mako default-fonts xdg-desktop-portal xdg-desktop-portal-gtk psmisc \
     mangohud \
     sdl2-compat ncurses \
+    procps-ng \
     dbus-daemon \
     "
 

--- a/images/base-app/build-fedora/configs/sway/config
+++ b/images/base-app/build-fedora/configs/sway/config
@@ -101,6 +101,8 @@ include /etc/sway/config.d/*
 
 bar swaybar_command waybar
 
+exec mako
+
 # App fixes
 # You can get more info about windows at runtime with `swaymsg -t get_tree`
 # Steam big picture fix (force windowing mode)

